test/integration/k8s_test.go

```diff
@@ -184,10 +184,34 @@ func TestPodScaling(t *testing.T) {
 				}
 				t.Log("all pods have been allocated IPs")
 			}) {
-				errors.New("Pods don't have IP's")
-				return
+				t.Error("some pods don't have IPs")
 			}
 
+			t.Run("nodenetworkconfig has converged", func(t *testing.T) {
+				clusterCheckCtx, cancel := context.WithTimeout(ctx, 20*time.Minute)
+				defer cancel()
+				nnccli, err := nodenetworkconfig.NewClient(restConfig)
+				if err != nil {
+					t.Fatal(err)
+				}
+				nodes, err := clientset.CoreV1().Nodes().List(clusterCheckCtx, metav1.ListOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				for _, node := range nodes.Items {
```

Comment:
This should run into a rangeVal lint exception that will get thrown:
https://go-critic.com/overview.html#rangeValCopy-ref

Reply:
Fair, but this is a test, so I'd prefer optimizing for readability so long as the tests are reasonably fast. The body of this loop is pretty IO-bound on the operations with the k8s API, so I think that would overwhelm the time spent copying and generating garbage.